### PR TITLE
Channel intrap

### PR DESCRIPTION
**Pull request by javier-m**

### CODE REVIEW

**Comment by javier-m** [comment]
WIP (need to add test)

File: tensorpipe/channel/intra_process/intra_process.cc [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+void IntraProcessChannelFactory::handleCopyRequests_() {
+  while (true) {
+    auto maybeRequest = requests_.pop();
+    if (!maybeRequest.has_value()) {
+      break;
+    }
+    CopyRequest request = std::move(maybeRequest).value();
+
+    // Perform copy.
+    auto nread = ::memcpy(request.localPtr, request.remotePtr, request.length);
```
Comment (beauby) [inline tensorpipe/channel/intra_process/intra_process.cc]: Use `std::memcpy` explicitly.

File: tensorpipe/channel/intra_process/intra_process.cc [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+void IntraProcessChannel::readProtoCallbackEntryPoint_(
+    const Error& error,
+    const proto::Packet& packet,
+    TBoundReadProtoCallback fn) {
+  if (processError(error)) {
+    return;
+  }
+  if (fn) {
```
Comment (beauby) [inline tensorpipe/channel/intra_process/intra_process.cc]: Can `fn` be unset?

**Comment by beauby** [comment]
Closing in favor of #174.

**Pull request closed without merging**